src/commands/moderation/prefixCommands/functions/removeChannelPermission.ts

```diff
@@ -0,0 +1,109 @@
+import { ChatInputCommandInteraction, Colors, TextChannel, User } from 'discord.js';
+import { constantsConfig, getConn, PrefixCommand, Logger, makeEmbed, refreshSinglePrefixCommandCache } from '../../../../lib';
+
+const noConnEmbed = makeEmbed({
+    title: 'Prefix Commands - Remove Channel - No Connection',
+    description: 'Could not connect to the database. Unable to remove the prefix command channel.',
+    color: Colors.Red,
+});
+
+const noCommandEmbed = (command: string) => makeEmbed({
+    title: 'Prefix Commands - Remove Channel - No Command',
+    description: `Failed to remove the prefix command channel for command ${command} as the command does not exist or there are more than one matching.`,
+    color: Colors.Red,
+});
+
+const failedEmbed = (command: string, channel: string) => makeEmbed({
+    title: 'Prefix Commands - Remove Channel- Failed',
+    description: `Failed to remove the prefix command channel <#${channel}> for command ${command}.`,
+    color: Colors.Red,
+});
+
+const doesNotExistEmbed = (command: string, channel: string) => makeEmbed({
+    title: 'Prefix Commands - Remove Channel - Does not exist',
+    description: `A prefix command channel <#${channel}> for command ${command} does not exist.`,
+    color: Colors.Red,
+});
+
+const successEmbed = (command: string, channel: string) => makeEmbed({
+    title: `Prefix command channel <#${channel}> removed for command ${command}.`,
+    color: Colors.Green,
+});
+
+const modLogEmbed = (moderator: User, command: string, channel: string) => makeEmbed({
+    title: 'Remove prefix command channel permission',
+    fields: [
+        {
+            name: 'Command',
+            value: command,
+        },
+        {
+            name: 'Channel',
+            value: `<#${channel}>`,
+        },
+        {
+            name: 'Moderator',
+            value: `${moderator}`,
+        },
+    ],
+    color: Colors.Red,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Remove Channel - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleRemovePrefixCommandChannelPermission(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const command = interaction.options.getString('command')!;
+    const channel = interaction.options.getChannel('channel')!;
+    const moderator = interaction.user;
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.